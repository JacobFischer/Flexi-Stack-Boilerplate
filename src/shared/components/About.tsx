import React from "react";

export const About = (): JSX.Element => (
    <>
        <h2>About</h2>
        <p><strong>Jacob Fischer</strong> made this page! It&apos;s simple boilerplate to get up and moving ASAP.</p>
    </>
);
