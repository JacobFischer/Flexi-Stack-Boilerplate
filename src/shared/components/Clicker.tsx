import React from "react";
import styled from "styled-components";

export const Button = styled.button({
    backgroundColor: "lightblue",
    border: "3px solid blue",
});

/**
 * An example component that maintains a state via React hooks.
 *
 * @returns A functional component that uses state.
 */
export function Clicker() {
    // const [count, setCount] = useState(0);
    const count = -1;

    return (
        <div>
            <p>You clicked me {count} times</p>
            <Button onClick={undefined /*() => setCount(count + 1)*/}>
                Click me
            </Button>
        </div>
    );
}
export default Clicker;
