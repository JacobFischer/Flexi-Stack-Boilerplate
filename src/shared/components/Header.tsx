import React from 'react';
import styled from 'styled-components';

const DarkBlue = styled.h1({
  color: 'darkblue',
  fontSize: '2rem',
  fontWeight: 'bold',
});

/**
 * A simple main Header.
 *
 * @param props - Props to spread.
 * @returns A functional react component.
 */
const Header = <P extends unknown>(props: React.PropsWithChildren<P>) => (
  <DarkBlue {...props} />
);

export default Header;
