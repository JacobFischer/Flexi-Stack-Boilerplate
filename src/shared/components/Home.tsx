import React from "react";

export const Home = (): JSX.Element => (
    <>
        <h1>Home</h1>
        <p>Welcome to the index page!</p>
    </>
);
