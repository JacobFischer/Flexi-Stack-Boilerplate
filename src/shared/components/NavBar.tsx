import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { route as routeAbout } from '../pages/about/route';
import { route as routeHome } from '../pages/home/route';

const List = styled.ul({
  backgroundColor: 'darkgreen',
});

const Item = styled.li({
  color: 'white',
  display: 'inline',
  listStyle: 'none',
  padding: '0 0.5rem',
});

const NavBar: React.FunctionComponent = () => (
  <List>
    <Item>
      <Link to={routeHome}>Home</Link>
    </Item>
    <Item>
      <Link to={routeAbout}>About</Link>
    </Item>
    <Item>
      <Link to="/nowhere">Nowhere</Link>
    </Item>
  </List>
);

export default NavBar;
