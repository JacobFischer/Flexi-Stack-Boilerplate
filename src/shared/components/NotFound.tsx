import React from "react";

/**
 * The page displayed when a route is not found to render, for 404s.
 *
 * @returns A react functional component.
 */
export const NotFound = () => (
    <>
        <h2>Not Found</h2>
        <pre>Sorry, but we couldn&apos;t find anything here!</pre>
    </>
);
