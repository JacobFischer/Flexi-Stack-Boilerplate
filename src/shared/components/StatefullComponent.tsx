import React, { useState } from "react";
import styled from "styled-components";

export const Button = styled.button({
    backgroundColor: "lightblue",
    border: "3px solid blue",
});

/**
 * An example component that maintains a state via React hooks.
 *
 * @returns A functional component that uses state.
 */
export function StatefullComponent() {
    const [count, setCount] = useState(0);

    return (
        <div>
            <p>You clicked me {count} times</p>
            <Button onClick={() => setCount(count + 1)}>Click me</Button>
        </div>
    );
}
