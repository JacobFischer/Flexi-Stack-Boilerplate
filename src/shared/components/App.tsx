import React from "react";
import { Link, Route, Switch } from "react-router-dom";
import styled from "styled-components";
import { routes } from "../routes";
import { NotFound } from "./NotFound";

const NavBar = styled.ul({
    backgroundColor: "red",
});

export const App = () => (
    <>
        <NavBar>
            {routes.map(([route]) => (<li key={route}><Link to={route}>{route}</Link></li>))}
            <li><Link to="/not-found">Dead link</Link></li>
        </NavBar>
        <Switch>
            {routes.map(([route, component]) => <Route key={route} exact path={route} component={component} />)}
            <Route component={NotFound} />
        </Switch>
    </>
);
