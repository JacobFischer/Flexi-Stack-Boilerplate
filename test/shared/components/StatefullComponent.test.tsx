import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import { Button, StatefullComponent } from "../../../src/shared/components/StatefullComponent";

describe("StatefullComponent component", () => {
    it("renders", () => {
        expect(renderer.create(<StatefullComponent />).toJSON()).toMatchSnapshot();
    });

    it("Counts state", () => {
        const wrapper = shallow(<StatefullComponent />);

        const button = wrapper.find(Button);
        expect(button.length).toBe(1);

        for (let i = 1; i < 9; i += 1) {
            button.simulate("click");
            // TODO: actually check once merged: https://github.com/airbnb/enzyme/issues/2011
            // expect(html.includes(String(i))).toBe(true);
        }
    });
});
