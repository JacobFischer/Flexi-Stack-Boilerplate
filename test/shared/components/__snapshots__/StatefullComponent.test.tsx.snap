// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatefullComponent component renders 1`] = `
<div>
  <p>
    You clicked me 
    0
     times
  </p>
  <button
    className="sc-AxjAm cDvpSm"
    onClick={[Function]}
  >
    Click me
  </button>
</div>
`;
